### PR TITLE
fix: migration gaslimit

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Set the withdrawal migration gas limit to
a non zero number.

### CODE REVIEW

File: op-chain-ops/crossdomain/migrate.go [hunk 1]
```diff
@@ -77,12 +79,15 @@ func MigrateWithdrawal(withdrawal *LegacyWithdrawal, l1CrossDomainMessenger *com
...
+	// Set the outer gas limit. This cannot be zero
+	gasLimit := uint64(len(data)*16 + 200_000)
+
 	w := NewWithdrawal(
 		withdrawal.Nonce,
 		&predeploys.L2CrossDomainMessengerAddr,
 		l1CrossDomainMessenger,
 		value,
-		new(big.Int),
+		new(big.Int).SetUint64(gasLimit),
```
Comment (trianglesphere) [inline op-chain-ops/crossdomain/migrate.go:90]: Is this the gas limit that is eventually passed to call which will be executed on L1?

Comment (tynes) [reply op-chain-ops/crossdomain/migrate.go:90]: Yup exactly

Comment (trianglesphere) [reply op-chain-ops/crossdomain/migrate.go:90]: Got it. I also just saw the doc you wrote up

File: op-chain-ops/crossdomain/migrate.go [hunk 2]
```diff
@@ -77,12 +79,15 @@ func MigrateWithdrawal(withdrawal *LegacyWithdrawal, l1CrossDomainMessenger *com
 		return nil, fmt.Errorf("cannot abi encode relayMessage: %w", err)
 	}
 
+	// Set the outer gas limit. This cannot be zero
+	gasLimit := uint64(len(data)*16 + 200_000)
```
Comment (protolambda) [inline op-chain-ops/crossdomain/migrate.go:83]: Would be good to document this + determine if it's a sufficient gas limit. LGTM otherwise.

Comment (tynes) [reply op-chain-ops/crossdomain/migrate.go:83]: We have a notion doc explaining the rationale, the plan is to merge and then test e2e

File: packages/sdk/src/cross-chain-messenger.ts [hunk 3]
```diff
@@ -351,10 +351,12 @@ export class CrossChainMessenger {
       }
     }
 
+    const minGasLimit = resolved.message.length * 16 + 200_000
```
Comment (smartcontracts) [inline packages/sdk/src/cross-chain-messenger.ts:354]: Message here isn't a buffer. You would need to convert to buffer first before getting length.

Comment (smartcontracts) [reply packages/sdk/src/cross-chain-messenger.ts:354]: Should be able to use `fromHexString`

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**